src/optim/abstraction/UniformGridAbstraction/optimal_control_problem.jl

```diff
@@ -1,3 +1,81 @@
+"""
+    OptimizerOptimalControlProblem{T} <: MOI.AbstractOptimizer
+
+An optimizer that solves reachability or reach-avoid **optimal control problems** using symbolic abstractions of the system.
+
+This solver takes as input a concrete problem (typically an instance of [`OptimalControlProblem`](@ref Dionysos.Problem.OptimalControlProblem)) and a symbolic abstraction of the system (i.e., an [`abstract_system`](@ref Dionysos.Symbolic.SymbolicModelList)). It then solves the **abstract** versions of the control problem.
+
+### Key Behavior
+
+- Lifts the concrete reachability problem to the symbolic abstraction space (`abstract_system`) and constructs the corresponding `abstract_problem`.
+- Computes the `controllable_set` — the largest set of abstract states from which reachability can be guaranteed.
+- Synthesizes a `abstract_controller` that brings the system within the target set under worst-case transitions.
+- Computes the `abstract_value_function` that maps each state (cell) to the worst-case number of steps needed to reach the target set.
+- The solver is successful if the field `success` is `true` after `MOI.optimize!`.
+
+### Parameters
+
+#### Inputs
+
+- `concrete_problem`: Instance of [`OptimalControlProblem`](@ref Dionysos.Problem.OptimalControlProblem), defining the reach-avoid control task.
+- `abstract_system`: The symbolic abstraction of the system, typically produced by an abstraction solver such as [`OptimizerEmptyProblem`](@ref Dionysos.Optim.Abstraction.UniformGridAbstraction.OptimizerEmptyProblem).
+
+#### Abstract Problem Fields
+
+- `abstract_problem`: The lifted problem expressed over the abstract system.
+- `abstract_controller`: Abstract controller.
+- `abstract_problem_time_sec`: Time taken to solve the abstract problem.
+
+#### Fixpoint Stopping
+
+- `early_stop`: A `Bool` that allows stopping the fixpoint iteration as soon as the initial set is fully contained in the growing **controllable set**. This is useful in reachability problems where the fixpoint expands the target set outward.
+
+#### Memory Optimization
+
+- `sparse_input`: If `true`, replaces the default state × input transition table with a sparse dictionary-based structure.  
+  This is useful when:
+  - The input space is large.
+  - Only a small subset of inputs are admissible per state.
+  - For example, in the determinized abstraction case where inputs are of the form `new_input = (input, target)`.
+
+#### Output Sets
+
+- `controllable_set`: The set of abstract states from which the target set can be reached (worst-case guaranteed).
+- `uncontrollable_set`: Complementary set of unreachable states under the chosen strategy.
+
+#### Value Functions
+
+- `value_fun_tab`: A tabular representation storing, for each state, the associated cost or number of steps to reach the target.
+- `abstract_value_function`: Maps abstract states (cells) to worst-case cost-to-go.
```

Comment:
Is that something set by user ? Is that mandatory ? Does it need to be exact or approximate ? Lower or upper bound ?

Reply:
It's computed by the solver. It just the worst-case number of stepsto reach the target set. I will clarify it.

Reply:
If it's computed by the solver, why is it here ?

Reply:
I clarified the nature of fields... like mandatory, user set, computed, ...